ls1intum/Artemis
# Feature/orion instructor view

### Checklist

### Motivation and Context


For the new instructor view of the IntelliJ plugin, we need some changes on the client side.
These include changes to the APIs between IntelliJ and Artemis as well as adaptions to the templates of the course administrator pages, so that the UI fits into the tool window in IntelliJ.

### Description

I added the missing API endpoints to the Java/JS bridge interfaces. **I also changed the `programming-exercise.component` and the `course.component`, so that they better fit onto smaller screens, meaning that we don't show columns like the presentation score, or the color of the course.**
For IntelliJ users, there are also some more adaptions (removed some buttons and columns that are not needed for the instructor view functionality)

I also removed even more services from barrel files. This is why there are so many changed files. In these cases, I only changed the import.

**As for the components that seem to be new:** I just refactored the code-editor-instructor component and created an inheritance hierarchy, so that we can either have the "regular" instructor editor, or the adapted one for IntelliJ, which only shows some parts of the whole editor. There is no real new logic in the client. All new calls just get routed into IntelliJ (Or for other calls, e.g. to ide-build-and-test-service I already wrote tests). That is also why I don't have any really new client tests, I just adapted the old ones. 

### Steps for Testing

Install the plugin: 
[orion-0.3.0-rc.2.zip](https://github.com/ls1intum/Artemis/files/4010304/orion-0.3.0-rc.2.zip)



1. Log in to Artemis (with the plugin)
2. Navigate to Course Administration
3. Select any course with a programming exercise
4. Click on "Edit in IntelliJ" in order to import the exercise (only Java for now)
5. All three base repositories should get cloned in one single project
5. You can select any of the repositories in the dropdown and click on submit or test locally
6. Testing locally, a Maven job should start that runs all tests and displays the result in the bottom console
7. Submitting commits and pushes all changes to the remote and displays the result of the remote tests in the bottom console

### Screenshots


![orion_instructor_demo](https://user-images.githubusercontent.com/27979674/69986475-c1fd1b00-153d-11ea-983d-b8683ed8ea39.gif)
